Klasyfikacja.Rmd

```diff
@@ -48,41 +47,40 @@ zapisie szesnastkowym, w trzeciej nazwa klasy.
 ## Operacje na rastrach
 
 Jak można zauważyć, nasze rastry różnią się rozdzielczością przestrzenną (scena 
-Landsat ma 30 m, natomiast klasy pokrycia terenu 10 m). Kiedy rastry posiadają
-różne rozdzielczości, to nie jest możliwe wykonywanie na nich operacji matematycznych.
-W takim przypadku musimy sprowadzić je do jednakowej rozdzielczości. Z racji iż
-proste metody przepróbkowania nie zwiększają ilości informacji przy upscalingu,
-to lepiej wykonać przepróbkowanie do niższej rozdzielczości.
-
-Przepróbkowanie (*resampling*) można wykonać przy pomocy funkcji `resample()`.
-Dostępne są różne metody przepróbkowania, ale w przypadku danych kategorycznych,
-koniecznie trzeba wykorzystać algorytm najbliższego sąsiada (`method = "near"`).
+Landsat ma 30 m, natomiast klasy pokrycia terenu 10 m).
+Kiedy rastry posiadają różne rozdzielczości nie jest możliwe wykonywanie na nich operacji matematycznych.
+W takim przypadku musimy sprowadzić je do jednakowej rozdzielczości.
+Z racji iż proste metody przepróbkowania (*resampling*) nie zwiększają ilości informacji przy upscalingu, to lepiej wykonać przepróbkowanie do niższej rozdzielczości.
+<!--JN: powyższe zdanie trzeba przepisać i trochę wyjaśnić-->
+
+Przepróbkowanie można wykonać przy pomocy funkcji `resample()`.
+Dostępne są różne metody przepróbkowania, ale w przypadku danych kategorycznych, koniecznie trzeba wykorzystać algorytm najbliższego sąsiada (`method = "near"`).
+<!-- JN: `mode` czy `med` też powinna zadziałać-->
```

Comment:
Z medianą to nie jestem pewien. Czy jeśli liczba pikseli będzie parzysta, to wtedy wyliczy średnią z dwóch środkowych wartości?

Reply:
Łatwo sprawdzić:

``` r
library(terra)
#> terra 1.6.38
set.seed(2022-11-06)
r1 = rast(ncols = 9, nrows = 9, vals = sample(1:2, size = 81, replace = TRUE))
r2 = rast(ncols = 5, nrows = 5)

resam1 = resample(r1, r2, method = "med")
resam1
#> class       : SpatRaster 
#> dimensions  : 5, 5, 1  (nrow, ncol, nlyr)
#> resolution  : 72, 36  (x, y)
#> extent      : -180, 180, -90, 90  (xmin, xmax, ymin, ymax)
#> coord. ref. : lon/lat WGS 84 
#> source(s)   : memory
#> name        : lyr.1 
#> min value   :     1 
#> max value   :     2

plot(resam1)
```

![](https://i.imgur.com/KbSvdSz.png)

``` r
freq(resam1)
#>   layer value count
#> 1     1     1    20
#> 2     1     2     5
```

<sup>Created on 2022-11-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Jednak w przypadku `aggregate()` to nie działa.

```r
library(terra)
set.seed(2022-11-06)
r1 = rast(ncols = 10, nrows = 10, vals = sample(1:2, size = 100, replace = TRUE))
r1 = as.factor(r1)
resam1 = aggregate(r1, fact = 2, fun = "median")
# freq(resam1) # zwraca złe wyniki
table(values(resam1))
#> 1 1.5   2 
#> 7  12   6 
```